### ISSUE TITLE
TCPError.accept upon launch

### ISSUE THREAD

**Issue by niksauer** [vapor/vapor#1451]
* Vapor Version: 3
* OS: macOS 
* OS Version: 10.13.12 (High Sierra)
* Vapor Toolbox Version: 3.1.3

### Expected Behavior
Application launches successfully with message: `Running RoadChatAPIv3 ... Server starting on http://localhost:8080`

### Current Behavior
Application is not accessible via http://localhost:8080 and launches with error message: 

```
[ ERROR ] Server Error: TCP.TCPError.accept: Invalid argument[ ERROR ] Server Error: TCP.TCPError.accept: Invalid argument[ ERROR ] Server Error: TCP.TCPError.accept: Invalid argument (TCPServer.swift: (TCPServer.swift: (TCPServer.swift:146)
[ ERROR ] Server Error: TCP.TCPError.accept: Invalid argument (TCPServer.swift:146)
146)
[ [ ERROR ] Server Error: TCP.TCPError.accept: Invalid argument (TCPServer.swift:146)
[ ERROR ] Server Error: TCP.TCPError.accept: Invalid argumentERROR ] Server Error: TCP.TCPError.accept: Invalid argument (TCPServer.swift:146)
[ ERROR ] Server Error: TCP.TCPError.accept: Invalid argument (TCPServer.swift (:146)
TCPServer.swift[ ERROR ] Server Error: TCP.TCPError.accept: Invalid argument (TCPServer.swift:146:146)
[ ERROR ] Server Error: TCP.TCPError.accept: Invalid argument[ ERROR ] Server Error: TCP.TCPError.accept: Invalid argument ()
TCPServer.swift:146)
[ ERROR ] Server Error: TCP.TCPError.accept: Invalid argument[ 146)
 ( (TCPServer.swift:146)TCPServer.swift:146
[ ERROR ] Server Error: TCP.TCPError.accept: Invalid argumentServer starting on ERRORhttp://localhost:8080
 ] Server Error: TCP.TCPError.accept: Invalid argument (TCPServer.swift:146)
 (TCPServer.swift:)
146)
```

### How to Reproduce
The application is running on Docker for Mac, with the following Dockerfile: 

```
FROM swift:4.0.3 as builder
WORKDIR /app/
COPY . .

RUN mkdir -p /build/lib && cp -R /usr/lib/swift/linux/*.so /build/lib
RUN swift build -c release && mv `swift build -c release --show-bin-path` /build/bin

FROM ubuntu:16.04

RUN \
  apt-get -q update && apt-get -q -y install \
  libatomic1 \
  libbsd0 \
  libcurl3 \
  libicu55 \
  libmysqlclient20 \
  libxml2;

RUN apt-get clean all;
RUN rm -r /var/lib/apt/lists/*;

WORKDIR /app/

COPY --from=builder /build/bin .
COPY --from=builder /build/lib/* /usr/lib/

EXPOSE 8080

CMD ["./Run"]
```

`docker build -t roadchatv3:latest .`
`docker run -p 8080:8080 --name roadchatv3 roadchatv3:latest`

The same configuration, with an additional `COPY --from=builder /app/Config ./Config` command, launches a default Vapor 2 project correctly.

**Comment by Joannis** [comment]
You can safely ignore this error. This error happens when the server is starting up and the worker threads are starting. The workers are trying to accept the first user(s) but are failing to do so because the socket isn't yet bound. After this the server should be running normally. One problem you may find is that the default port for production is currently `80`, so that would explain the problems connecting to your server. You can always override the port using the CLI.

**Comment by Joannis** [comment]
I've marked this a bug because the fake error messages should be silenced.

**Comment by niksauer** [comment]
After following this great [tutorial](https://bygri.github.io/2018/01/25/vapor-3-with-docker.html) a connection has been made possible again (albeit with the same fake error message)! Most importantly, `--hostname 0.0.0.0` may not be omitted from `swift run Run serve --hostname 0.0.0.0`.

Any specific reason, as to why it is not auto-configured like in v2?

**Comment by Joannis** [comment]
@niksauer this has been an ongoing debate in our slack. Binding to localhost is a "secure by default" kind of setting where you cannot accidentally expose a service to the wrong ports. A good example of this is MongoDB where some packages bound to `0.0.0.0` by default, whilst MongoDB doesn't need authentication by default. This resulted in significant leaks. So it's better to be safe than sorry, is the reasoning.

On the other hand, if you're deploying a (web) service on a host connected to the public internet (no firewall in between) it is probably not unreasonable that you have security holes in your application. So the convenience might outweight the leaks that occur due to deployment faults.

#1455 is closely related to this discussion.

**Comment by tanner0101** [comment]
Fixed by #1532
